tests/LoggerTest.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace OpenEuropa\EPoetry\Tests;
+
+use OpenEuropa\EPoetry\Tests\Logger\MockLogger;
+use OpenEuropa\EPoetry\Type\ContactPersonIn;
+use OpenEuropa\EPoetry\Type\Contacts;
+use OpenEuropa\EPoetry\Type\CreateRequests;
+use OpenEuropa\EPoetry\Type\LinguisticRequestIn;
+use OpenEuropa\EPoetry\Type\RequestGeneralInfoIn;
+use Psr\Log\LogLevel;
+
+/**
+ * @internal
+ * @coversNothing
+ */
+final class LoggerTest extends AbstractTest
+{
+    /**
+     * Test logging only error events.
+     */
+    public function testErrorLogging()
+    {
+        // Generate request to be sent.
+        $createRequests = $this->generateRequest();
+
+        $exception = new \Exception('Request was not properly formatted.');
+        $this->httpClient->addException($exception);
+
+        $logger = new MockLogger();
+        $clientFactory = $this->createClientFactory();
+        $clientFactory->setLogger($logger);
+        $clientFactory->setLogLevel(LogLevel::ERROR);
+        $client = $clientFactory->getClient();
+
+        try {
+            $client->createRequests($createRequests);
+        } catch (\Exception $e) {
+            // Do nothing.
+        }
```

Comment:
We should not throw exceptions in tests, tests should fail if any exception is thrown by code that their run.

Reply:
In this case we need the createRequest call to throw exceptions so they are logged as errors

Reply:
Ah right! ok then it's fine of course! :smile: